sklearn/semi_supervised/_self_training.py

```diff
@@ -237,6 +253,12 @@ def fit(self, X, y):
                 UserWarning,
             )
 
+        if _routing_enabled():
+            routed_params = process_routing(self, "fit", **params)
+        else:
+            routed_params = Bunch()
+            routed_params.base_estimator = Bunch(fit={})
```

Comment:
I think it makes sense to rename `base_estimator` to `estimator` at the same time too.